packages/accessibility/src/behaviors/Button/buttonBehavior.ts

```diff
@@ -9,26 +9,37 @@ import { Accessibility } from '../../types'
  * Adds attribute 'aria-disabled=true' based on the property 'loading'.
  * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.
  */
-const buttonBehavior: Accessibility<ButtonBehaviorProps> = props => ({
-  attributes: {
-    root: {
-      role: props.as === 'button' ? undefined : 'button',
-      tabIndex: props.as === 'button' ? undefined : 0,
-      'aria-disabled': props.disabled || props.loading,
+const buttonBehavior: Accessibility<ButtonBehaviorProps> = props => {
+  const definition: AccessibilityDefinition = {
+    attributes: {
+      root: {
+        role: props.as === 'button' ? undefined : 'button',
+        tabIndex: props.as === 'button' ? undefined : 0,
+        disabled:
+          props.disabled && !props.loading ? (props.as === 'button' ? true : undefined) : undefined,
```

Comment:
It will be disabled, but with aria-disabled only, not with the native `disabled` attribute. Using native attributes is more preferable in order to have the default native logic of setting an attribute. But because of that native logic we need this condition for the loading button case for the sake of backwards compatibility. If you disable a currently focused button, it will lose the focus. At the same time, we have this loading button (I would propose to make a separate component named LoadingButton, by the way, because Button is probably the most used component and the loading button overhead is something which is almost never used) and it shouldn't lose the focus (because it usually gets enabled back quickly enough and we don't want to have a dedicated logic for moving the focus back). So we have this condition, and there are two classes in the schema. One is for the standard button and it requires `<button>` to be disabled with the `disabled` attribute, and there is another one for the loading button.